### PR TITLE
ci: fix e2e tests for pnpm and node v20

### PR DESCRIPTION
**Pull request by sshelomentsev**

### CODE REVIEW

**Review by TheUnderScorer — DISMISSED** [review]

File: .github/workflows/deploy_test_branch.yml [hunk 1]
```diff
@@ -20,24 +20,17 @@ jobs:
         with:
           node-version-file: '.node-version'
 
+      - name: Install pnpm
+        run: npm i -g pnpm
```
Comment (ilfa) [inline .github/workflows/deploy_test_branch.yml:24]: Let`s use cache like we do in [dx-toolkit](https://github.com/fingerprintjs/dx-team-toolkit/blob/main/.github/workflows/build-typescript-project.yml#L63)

And also we can use github action `pnpm/action-setup` to install pnpm (but we should do it before installing node)

**Review by ilfa — CHANGES_REQUESTED** [review]
Can [pnpm workspaces](https://pnpm.io/workspaces#workspace-protocol-workspace) help us to simplify some steps?

**Review by TheUnderScorer — APPROVED** [review]

**Pull request merged**